charms/istio-pilot/src/charm.py

```diff
@@ -187,29 +192,36 @@ def install(self, _):
         global_proxy_image = image_config["global-proxy-image"]
         global_proxy_init_image = image_config["global-proxy-init-image"]
 
-        # Call istioctl install and set parameters based on image configuration
-        subprocess.check_call(
-            [
-                "./istioctl",
-                "install",
-                "-y",
-                "--set",
-                "profile=minimal",
-                "--set",
-                f"values.global.istioNamespace={self.model.name}",
-                "--set",
-                f"values.pilot.image={pilot_image}",
-                "--set",
-                f"values.global.tag={global_tag}",
-                "--set",
-                f"values.global.hub={global_hub}",
-                "--set",
-                f"values.global.proxy.image={global_proxy_image}",
-                "--set",
-                f"values.global.proxy_init.image={global_proxy_init_image}",
-            ]
+        # Generate extra flags to pass to the istioctl install command
+        istioctl_extra_flags = [
+            "--set",
+            f"values.pilot.image={pilot_image}",
+            "--set",
+            f"values.global.tag={global_tag}",
+            "--set",
+            f"values.global.hub={global_hub}",
+            "--set",
+            f"values.global.proxy.image={global_proxy_image}",
+            "--set",
+            f"values.global.proxy_init.image={global_proxy_init_image}",
+        ]
+
+        # Call the istioctl wrapper to install the Istio Control Plane
+        istioctl = Istioctl(
+            ISTIOCTL_PATH,
+            self.model.name,
+            ISTIOCTL_DEPOYMENT_PROFILE,
+            istioctl_extra_flags=istioctl_extra_flags,
         )
 
+        try:
+            istioctl.install()
+        except IstioctlError as e:
+            self.log.error(INSTALL_FAILED_MSG.format(message="istioctl install failed"))
```

Comment:
Any reason why we don't use `message=str(e)` so we provide the error message that Istioctl creates?  I think it has a bit more information, and is the same as how other errors are passed to the user